evergreen.yml

```diff
@@ -292,6 +315,18 @@ task_groups:
     tasks:
       - test-rust
 
+  - name: test-result-set-group
```

Comment:
I renamed it because Patrick's connection PR added a `test-integration-group` [here](https://github.com/mongodb/mongo-odbc-driver/blob/e8dcba32f8afe87ee9ddc521a3522bcc2a2347c4/evergreen.yml#L296), which was the conflict I mentioned during standup.   
One idea could be to run them together in the same task.  For now it might be good to run them separately, as that may help with troubleshooting them.  `test-result-set-group` was the first descriptive name that came to mind, how about`test-harness-group`, `test-query-function-group`?

Reply:
I'll keep the current name and we can modify it in a future PR if necessary.